lib/interfaces/appstore.js

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright 2017 Scott Bender <scott@scottbender.net>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+const debug = require("debug")("signalk:interfaces:appstore");
+const _ = require("lodash");
+const fs = require("fs");
+const agent = require("superagent-promise")(require("superagent"), Promise);
+const compareVersions = require("compare-versions");
+const spawn = require("child_process").spawn;
+
+module.exports = function(app) {
+  return {
+    start: function() {
+      app.availablePlugins = [];
```

Comment:
I am not too crazy about these stashed in the app. You could define them as regular variables before the return statement above and define the functions that need them in the same scope.